hudi-common/src/main/java/org/apache/hudi/common/model/RecordPayloadType.java

```diff
@@ -98,12 +97,10 @@ public static String getPayloadClassName(HoodieConfig config) {
     String payloadClassName;
     if (config.contains(PAYLOAD_CLASS_NAME)) {
       payloadClassName = config.getString(PAYLOAD_CLASS_NAME);
-    } else if (config.contains(PAYLOAD_TYPE)) {
-      payloadClassName = RecordPayloadType.valueOf(config.getString(PAYLOAD_TYPE)).getClassName();
     } else if (config.contains("hoodie.datasource.write.payload.class")) {
       payloadClassName = config.getString("hoodie.datasource.write.payload.class");
     } else {
-      payloadClassName = RecordPayloadType.valueOf(PAYLOAD_TYPE.defaultValue()).getClassName();
+      return DefaultHoodieRecordPayload.class.getName();
```

Comment:
Use the defined default legacy payload class variable here instead of hardcoding the value.

Reply:
Fixed.